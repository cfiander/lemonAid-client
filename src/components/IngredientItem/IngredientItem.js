import React from 'react'
import RecipeContext from '../../contexts/RecipeContext'




export default class Ingredient extends React.Component {
  static defaultProps ={
    handleClickDelete: () => {},
  }
  static contextType = RecipeContext

  handleClickDelete = e => {
    e.preventDefault()
    const ingredientId = this.props.id
    this.context.deleteIngredient(ingredientId) 
  }

  render() {
    const { name, id, } = this.props
    return (
      <>
        <h2>
          {name}
        </h2>
        <div className="lemonLogo">&#127819;</div>
        <button
          className="remove basicButton"
          type='button'
          onClick={this.handleClickDelete}
        >
          REMOVE
        </button>
      </>
    )
  }
}