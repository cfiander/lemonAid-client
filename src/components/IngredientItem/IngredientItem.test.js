import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import IngredientItem from './IngredientItem'


describe(`IngredientItem component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
  }

  it('renders an ingredient by default', () => {
    const wrapper = shallow(<IngredientItem />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders an ingredient given props', () => {
    const wrapper = shallow(<IngredientItem {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})