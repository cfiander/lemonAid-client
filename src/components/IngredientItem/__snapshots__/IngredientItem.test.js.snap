// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientItem component renders an ingredient by default 1`] = `
<div
  className="asdf"
>
  <h2
    className="asdf"
  />
  <button
    className="addIngredientButton"
    onClick={[Function]}
    type="button"
  >
    REMOVE
  </button>
</div>
`;

exports[`IngredientItem component renders an ingredient given props 1`] = `
<div
  className="asdf"
  key="a"
>
  <h2
    className="asdf"
  >
    test-class-name
  </h2>
  <button
    className="addIngredientButton"
    onClick={[Function]}
    type="button"
  >
    REMOVE
  </button>
</div>
`;
